src/services/webapi/webapi.service.factory.ts

```diff
@@ -2,13 +2,25 @@ import { IWizdomContext } from "../context/context.interfaces";
 import { IWizdomWebApiService, IWizdomWebApiServiceState } from "./webapi.interfaces";
 import { WizdomWebApiService } from "./webapi.service";
 import { IWizdomCorsProxyServiceFactory } from "../corsproxy/corsproxy.interfaces";
+import { WizdomAADWebApiService } from "./aad-webapi.service";
+import { AadHttpClient } from "@microsoft/sp-http";
 
 export class WizdomWebApiServiceFactory {
-    constructor(private corsProxyFactory: IWizdomCorsProxyServiceFactory, private spHostUrl: string) {                
+    private spHostUrl: string;
+    constructor(private corsProxyFactory: IWizdomCorsProxyServiceFactory, private wizdomContext: IWizdomContext, private spContext: any, private aadHttpClientPromise: Promise<AadHttpClient>) {                
```

Comment:
Im not a big fan of passing the spContext object, just to extract the spHostUrl here. Instead just pass the spHostUrl, as in the original implementation

Reply:
I honestly can't remember why I did that, I think I had an earlier version where I was using more than just the SPHostUrl and then never re-refactored it back - I think maybe the aad-webapi service needed the spContext in one if the early implementations and that's why, but as I see it there's no problem switching it back to spHostUrl now